src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -1055,15 +1060,57 @@ emit_vector64_vector128_t (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 		break;
 	}
 	case SN_op_Addition:
-	case SN_op_Subtraction: {	
+	case SN_op_Subtraction:
+	case SN_op_Division:
+	case SN_op_Multiply:
+	case SN_op_BitwiseAnd:
+	case SN_op_BitwiseOr:
+	case SN_op_ExclusiveOr: {
 		if (!(fsig->param_count == 2 && mono_metadata_type_equal (fsig->ret, type) && mono_metadata_type_equal (fsig->params [0], type) && mono_metadata_type_equal (fsig->params [1], type)))
 			return NULL;
 		MonoInst *ins = emit_simd_ins (cfg, klass, OP_XBINOP, args [0]->dreg, args [1]->dreg);
 		ins->inst_c1 = etype->type;
-		if (etype->type == MONO_TYPE_R4 || etype->type == MONO_TYPE_R8)
-			ins->inst_c0 = id == SN_op_Addition ? OP_FADD : OP_FSUB;
-		else
-			ins->inst_c0 = id == SN_op_Addition ? OP_IADD : OP_ISUB;
+
+		if (etype->type == MONO_TYPE_R4 || etype->type == MONO_TYPE_R8) {
+			switch (id) {
+			case SN_op_Addition:
+				ins->inst_c0 = OP_FADD;
+				break;
+			case SN_op_Subtraction:
+				ins->inst_c0 = OP_FSUB;
+				break;
+			case SN_op_Multiply:
+				ins->inst_c0 = OP_FMUL;
+				break;
+			case SN_op_Division:
+				ins->inst_c0 = OP_FDIV;
+				break;
+			default:
+				NULLIFY_INS (ins);
+				return NULL;
+			}
+		} else {
```

Comment:
You haven't implemented integer multiplication and division. Is there a reason for that? Will those be part of a different PR?

Reply:
In all places divide and division (method, operators) are not implemented for integer so I decided to do the same. In emit_sys_numerics_vector_t 'SN_op_Division' and 'SN_op_Multiply' are commented for integer so not sure if there is a reason for it or no.

Reply:
For `Division` method, I only found intrinsics for non-integer types. For `Multiply` method, there is intrinsics for all types.

Reply:
Right, there is no direct SIMD acceleration support for integer division, just multiply.

You could accelerate a few cases of division by constant, but that's not something that has been a focus for the JIT yet.